app/Main.hs

```diff
@@ -222,11 +228,11 @@ runCommand options env =
     -- replaces the current process with `command`. It does not return.
     executeFile command searchPath args env'
 
-
-requestSecret :: Options -> Secret -> IO (Either VaultError EnvVar)
-requestSecret opts secret =
+-- | Requests all the data associated with a secret from the vault.
```

Comment:
Nit: Please use imperative case for comments. I.e. `Request all data associated...`